plugins/chromatic_correction.py

```diff
@@ -0,0 +1,392 @@
+                   "3. No two streams should be the same.")
+            self.dlg.lbl_description.SetLabel(txt)
+            self.dlg.addButton("Manually fix")
+            self.dlg.ShowModal()
+            return
+        self.dlg.Bind(wx.EVT_CLOSE, self.on_close)
+        self.dlg.addButton("Start", self._on_start, face_colour='blue')
+        self.dlg.ShowModal()
+
+    def on_close(self, evt):
+        self.dlg.buttons[0].SetLabel("Cancelling...")
+        self.dlg.buttons[0].Disable()
+        self.dlg.lbl_description.SetLabel("Cancellation in progress\nMay take some time to "
+                                          "cancel the already running processes")
+        self.cancelled = True
+
+    def _on_start(self, evt):
+        """Start or cancel the correction method when the button is clicked"""
+        # If correction is running, cancel it, otherwise start one
+        if self.dlg.buttons[0].Label == "Start":
+            self.cancelled = False
+            self._start_chromatic_correction()
+        else:
+            self.on_close(evt)
+
+    def _start_chromatic_correction(self):
+        # Change the Start button to Cancel
+        self.dlg.buttons[0].SetLabel("Cancel")
+        localization_tab = self.main_app.main_data.getTabByName("cryosecom-localization")
+        tab_data = localization_tab.tab_data_model
+        self.dlg.lbl_description.SetLabel("In Progress\nTaking Z stack of the given channels")
+        mip_das_dict = {}  # keys are filter band positions and values are mip data arrays
+        # Acquire the Maximum Intensity Projection for the given streams
+        levels = generate_zlevels(tab_data.main.focus, (-10e-06, 10e-06), 10e-06)
+        streams_list = [s for s in tab_data.streams.value
+                        if isinstance(s, FluoStream)]
+        zlevels = {stream: levels for stream in streams_list}
+        settings_observer = tab_data.main.settings_obs
+        acqui_task = acquireZStack(streams_list, zlevels, settings_obs=settings_observer)
+
+        # Check at regular intervals if the correction is cancelled by the user
+        if self.cancelled:
+            self.dlg.Destroy()
+            return
+
+        da_list, _ = acqui_task.result()
+        ref_em_pos = None
+
+        for da in da_list:
+            mip_image = model.DataArray(numpy.amax(da, axis=0), copy.copy(da.metadata))
+            file_location = get_file_location(f"{da.metadata[model.MD_DESCRIPTION]}", "ome.tiff")
+            dataio.tiff.export(file_location, mip_image)
+            if all(numpy.isclose(numpy.array(da.metadata[model.MD_OUT_WL]), self.ref_emission, rtol=1e-09, atol=1e-09)):
+                ref_em_pos = streams_list[0]._emission_to_idx[tuple(da.metadata[model.MD_OUT_WL])]
+                mip_das_dict[ref_em_pos] = mip_image
+            else:
+                em_index = streams_list[0]._emission_to_idx[tuple(da.metadata[model.MD_OUT_WL])]
+                mip_das_dict[em_index] = mip_image
```

Comment:
I think this would be easier to read:
```suggestion
            em_index = streams_list[0]._emission_to_idx[tuple(da.metadata[model.MD_OUT_WL])]
            mip_das_dict[em_index] = mip_image
            if all(numpy.isclose(numpy.array(da.metadata[model.MD_OUT_WL]), self.ref_emission, rtol=1e-09, atol=1e-09)):
                ref_em_pos = em_index

```